File: encapsulation.cpp
#include<bits/stdc++.h>
using namespace std;

class Bank{
    public:
        string name;
        int acc_no;

        Bank(string name, int acc_no){
        this->name = name;
        this->acc_no = acc_no;
        // this->balance = balance;
        }

    private:
        int balance;

    protected:
        string password;

    
};

int main(){

    // ific.name = "sumon debnath";
    // ific.acc_no = 101;
    Bank ific("Sumon Debnath", 101);

    // ific.password = "123456";        // given error for access protected and private values

    cout<<ific.name<<" "<<ific.acc_no<<endl;
    
    // cout<<ific.password<<endl;       // given error for print protected and private values

    return 0;
}

File: encapsulation.py

# copied by phitron oop class.

class Bank:
    def __init__(self, name, deposit):
        self.name = name                        # public
        self._branch = 'badda'                  # protected 
        self.__balance = deposit                # private

    def deposit(self, amount):
        self.__balance += amount

    def get_balance(self):
        return self.__balance
    
    def withdraw(self, amount):
        if amount < self.__balance:
            self.__balance = self.__balance - amount
            return amount
        else:
            return f'Forkia taka nai'

sumon = Bank('Sumon Debnath', 100)
sumon.deposit(300)
print(sumon.get_balance())